File: 1491-number-of-times-binary-string-is-prefix-aligned/1491-number-of-times-binary-string-is-prefix-aligned.py
class Solution(object):
    def numTimesAllBlue(self, flips):
        """
        :type flips: List[int]
        :rtype: int
        """
        max_on =0
        shining=0

        for i, bulb in enumerate(flips):
            max_on=max(max_on, bulb)
            if(max_on==i+1):
                shining+=1
        return shining
        

File: 1491-number-of-times-binary-string-is-prefix-aligned/1491-number-of-times-binary-string-is-prefix-aligned.java
class Solution {
    public int numTimesAllBlue(int[] nums) {
        int count = 0; // Counts the moments when all turned-on bulbs shine
        int maxOn = 0; // Tracks the maximum bulb index turned on

        for (int i = 0; i < nums.length; i++) {
            maxOn = Math.max(maxOn, nums[i]); // Update the maximum bulb index turned on
            
            // If the maximum bulb index equals the current moment (i + 1),
            // it means all bulbs up to this moment are on and shining
            if (maxOn == i + 1) {
                count++;
            }
        }

        return count;
    }
}
